trio/_timeouts.py

```diff
@@ -1,10 +1,11 @@
 import math
 from contextlib import contextmanager
+from typing import Iterator, ContextManager
```

Comment:
Hm. You're probably aware, but as of [Python 3.9](https://peps.python.org/pep-0585/), `collections.abc` is to be preferred. I'm decently sure it's fine to stick to `typing` versions for now and use [pyupgrade](https://github.com/asottile/pyupgrade) when dropping 3.8 so that we get the new non-deprecated syntax?

Reply:
Speaking of which can you run `pyupgrade` on your PR dropping 3.7? It might find some stuff!

Reply:
I changed them out - and might stick to collections.abc/contextlib mostly for the sake of getting used to the habit, but yeah don't think it really matters.

Reply:
There doesn’t need to be any rush to switch over - `typing` isn’t planning on removing these for a long while, given they’re just aliases.

Reply:
Oh yeah maybe I wasn't completely clear: I'm aok on using `typing` versions especially since there's a tool that will automatically switch them out, just as long as we're aware and we run that tool when we drop 3.8.